ArchUnitNET/Loader/LoadTasks/AddMembers.cs

```diff
@@ -68,8 +68,15 @@ private IMember CreatePropertyMember(PropertyDefinition propertyDefinition)
             var isStatic = (propertyDefinition.SetMethod != null && propertyDefinition.SetMethod.IsStatic) ||
                            (propertyDefinition.GetMethod != null && propertyDefinition.GetMethod.IsStatic);
             bool? isReadOnly = (propertyDefinition.SetMethod == null);
+            // HACK: Init setters are C# 8 feature that is only available in .net 5 and later.
+            // ArcUnitNET does not currently target those versions.
+            // That is why ModifierType.FullName is checked since there is no access to the IsExternalInit type.
+            bool isInitSetter = propertyDefinition.SetMethod != null
+                && propertyDefinition.SetMethod.ReturnType.IsRequiredModifier
+                && ((RequiredModifierType)propertyDefinition.SetMethod.ReturnType).ModifierType.FullName
+                    == "System.Runtime.CompilerServices.IsExternalInit";
```

Comment:
Instead of using comments, I would prefer to extract the logic into a method with a descriptive name, like `CheckPropertyHasInitSetterInNetStandardCompatibleWay`